From 898b30a183dedc49595f66d1bb8e0f18e5318687 Mon Sep 17 00:00:00 2001
From: Steven Moreland <smoreland@google.com>
Date: Thu, 20 Apr 2017 12:56:27 -0700
Subject: [PATCH 04/47] libwebm: Android.mk -> Android.bp

Test: all of libvpx links.
Change-Id: Iaa50db4be5a5e2bc0aa26c560db47f5802f07e43
Merged-Id: Iaa50db4be5a5e2bc0aa26c560db47f5802f07e43
(cherry picked from commit 5280486baa0424f5d480c2fdefab5f28f614b932)
---
 Android.bp         |  3 +++
 Android.mk         |  2 --
 libwebm.mk         | 11 -----------
 libwebm/Android.bp |  5 +++++
 4 files changed, 8 insertions(+), 13 deletions(-)
 create mode 100644 Android.bp
 delete mode 100644 libwebm.mk
 create mode 100644 libwebm/Android.bp

diff --git a/Android.bp b/Android.bp
new file mode 100644
index 0000000..927e1b9
--- /dev/null
+++ b/Android.bp
@@ -0,0 +1,3 @@
+subdirs = [
+    "libwebm",
+]
diff --git a/Android.mk b/Android.mk
index d0fe10a..b9da190 100644
--- a/Android.mk
+++ b/Android.mk
@@ -5,5 +5,3 @@ include $(CLEAR_VARS)
 # if ARMv7 + NEON etc blah blah
 include external/libvpx/libvpx.mk
 
-# libwebm
-include external/libvpx/libwebm.mk
diff --git a/libwebm.mk b/libwebm.mk
deleted file mode 100644
index 3afa4a0..0000000
--- a/libwebm.mk
+++ /dev/null
@@ -1,11 +0,0 @@
-# libwebm
-LOCAL_PATH := $(call my-dir)
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := libwebm/mkvparser/mkvparser.cc
-LOCAL_CPP_EXTENSION := .cc
-LOCAL_C_INCLUDES += $(LOCAL_PATH)/libwebm/
-
-LOCAL_MODULE := libwebm
-
-include $(BUILD_STATIC_LIBRARY)
diff --git a/libwebm/Android.bp b/libwebm/Android.bp
new file mode 100644
index 0000000..c770c49
--- /dev/null
+++ b/libwebm/Android.bp
@@ -0,0 +1,5 @@
+cc_library_static {
+    name: "libwebm",
+    srcs: ["mkvparser/mkvparser.cc"],
+    export_include_dirs: ["."],
+}
-- 
2.17.1

